internal/cli/cmd/cluster/build.go

```diff
@@ -203,111 +223,305 @@ func NewBuildCmd() *cobra.Command {
 	push := cmd.Flags().Bool("push", false, "If specified, pushes the image to the target repository.")
+	dockerLoad := cmd.Flags().Bool("load", false, "If specified, load the image to the local docker registry.")
 	tags := cmd.Flags().StringSliceP("tag", "t", nil, "Attach a tags to the image.")
+	platforms := cmd.Flags().StringSlice("platform", []string{platformAmd64}, "Set target platform for build")
 
 	cmd.RunE = fncobra.RunE(func(ctx context.Context, specifiedArgs []string) error {
+		// XXX: having multiple outputs is not supported by buildctl.
+		if *push && *dockerLoad {
+			return fnerrors.New("usage of both --push and --load flags is not supported")
+		}
 
 		buildctlBin, err := buildctl.EnsureSDK(ctx, host.HostPlatform())
 		if err != nil {
 			return fnerrors.New("failed to download buildctl: %w", err)
 		}
 
-		p, err := runBuildProxy(ctx)
+		clusterProfiles, err := api.GetProfile(ctx, api.Endpoint)
 		if err != nil {
 			return err
 		}
 
-		defer p.Cleanup()
-
 		contextDir := "."
 		if len(specifiedArgs) > 0 {
 			contextDir = specifiedArgs[0]
 		}
 
-		args := []string{
-			"build",
-			"--frontend=dockerfile.v0",
-			"--local", "context=" + contextDir,
-			"--local", "dockerfile=" + contextDir,
-		}
-
-		if *dockerFile != "" {
-			args = append(args, "--opt", "filename="+*dockerFile)
-		}
+		reindexer := newImageReindexer(ctx)
+		builtImages := make(map[string][]specs.Platform)
 
-		var complete func() error
+		errc := make(chan error, len(*platforms))
+		var wg sync.WaitGroup
+		for _, p := range *platforms {
+			platformSpec, err := platform.ParsePlatform(p)
+			if err != nil {
+				return err
+			}
 
-		if *push {
-			imageNames := []string{}
+			var imageNames []string
 			for _, tag := range *tags {
 				parsed, err := name.NewTag(tag)
 				if err != nil {
 					return fmt.Errorf("invalid tag %s: %w", tag, err)
 				}
 
-				imageNames = append(imageNames, parsed.Name())
+				builtImages[parsed.Name()] = append(builtImages[parsed.Name()], platformSpec)
+				imageNames = append(imageNames, imageNameWithPlatform(parsed.Name(), platformSpec))
 			}
 
-			args = append(args,
-				// buildctl parses output as csv; need to quote to pass commas to `name`.
-				"--output", fmt.Sprintf("type=image,push=true,%q", "name="+strings.Join(imageNames, ",")),
-			)
-
-			complete = func() error {
-				fmt.Fprintf(console.Stdout(ctx), "Pushed:\n")
-				for _, imageName := range imageNames {
-					fmt.Fprintf(console.Stdout(ctx), "  %s\n", imageName)
-				}
-
-				return nil
-			}
-		} else {
-			f, err := os.CreateTemp("", "docker-image-nsc")
+			buildProxy, err := runBuildProxy(ctx, resolveBuildCluster(p, clusterProfiles.ClusterPlatform))
 			if err != nil {
 				return err
 			}
+			defer buildProxy.Cleanup()
+			reindexer.setRegistryAccess(buildProxy.RegistryEndpoint, oci.RegistryAccess{Keychain: nscloud.DefaultKeychain})
+
+			wg.Add(1)
+			go func(platform string) {
```

Comment:
In case tenant only has access to amd64 builder, is it possible to only issue one `buildctl` command for both architectures? 

With `--opt platform=linux/amd64,linux/arm64` flag:
https://github.com/moby/buildkit/blob/master/docs/multi-platform.md

Reply:
Yes, it is possible, but this would require to a bit of refactoring, e.g we do not need to do reindexing. Tbh, I would not do it for now and just have two parallel builds.

Reply:
SG, it would also need to make sure that reindexing is consistent between the two cases.